fix: Bring back error output for links

mre opened the pull request:
Superseded by #1553

---


With the last lychee release, we simplified the status output for links.

While this reduced the visual noise, it also accidentally caused the source of errors to not be printed anymore. This change brings back the additional error information as part of the final report output. Furthermore, it shows the error information in the progress output if verbose mode is activated.

Fixes #1487

mre committed:
fix: Bring back error output for links

With the last lychee release, we simplified the status output for links.  While
this reduced the visual noise, it also accidentially caused the source of
errors to not be printed anymore. This change brings back the additional error
information as part of the final report output. Furthermore, it shows the error
information in the progress output if verbose mode is activated.

mre committed:
Cleanup outputs; update tests

mre committed:
cleanup

dmathieu committed:
Allow excluding cache based on status code (#1403)

This introduces an option `--cache-exclude-status`, which allows specifying a range of HTTP status codes which will be ignored from the cache.

Closes #1400.

dependabot[bot] committed:
Bump the dependencies group with 3 updates (#1530)

Bumps the dependencies group with 3 updates: [clap](https://github.com/clap-rs/clap), [secrecy](https://github.com/iqlusioninc/crates) and [octocrab](https://github.com/XAMPPRocky/octocrab).


Updates `clap` from 4.5.19 to 4.5.20
- [Release notes](https://github.com/clap-rs/clap/releases)
- [Changelog](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/clap-rs/clap/compare/clap_complete-v4.5.19...clap_complete-v4.5.20)

Updates `secrecy` from 0.10.2 to 0.10.3
- [Commits](https://github.com/iqlusioninc/crates/commits)

Updates `octocrab` from 0.41.0 to 0.41.1
- [Release notes](https://github.com/XAMPPRocky/octocrab/releases)
- [Changelog](https://github.com/XAMPPRocky/octocrab/blob/main/CHANGELOG.md)
- [Commits](https://github.com/XAMPPRocky/octocrab/compare/v0.41.0...v0.41.1)

---
updated-dependencies:
- dependency-name: clap
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: dependencies
- dependency-name: secrecy
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: dependencies
- dependency-name: octocrab
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mondeja committed:
Update `pkg-url` of cargo binstall (#1532)

thomas-zahner commented on lychee-bin/src/formatters/response/color.rs:
```diff
@@ -10,52 +10,75 @@ use super::{ResponseFormatter, MAX_RESPONSE_OUTPUT_WIDTH};
     use lychee_lib::{ErrorKind, Status, Uri};
+    use pretty_assertions::assert_eq;
+
+    #[cfg(test)]
```
What is the purpose of this annotation? Isn't it redundant, since the whole module `tests` is already marked as such?

mre replied:
You're right. It's unnecessary.

thomas-zahner commented on lychee-bin/src/formatters/response/color.rs:
```diff
@@ -65,14 +88,18 @@ mod tests {
+    #[test]
+    fn test_format_status() {
+        let status = Status::Ok(StatusCode::OK);
+        assert_eq!(ColorFormatter::format_status(&status).trim_start(), "[200]");
```
here you do `trim_start()` while in the below tests you don't. Is there a reason for that? Shouldn't we test whitespace here as well?

mre replied:
Good point. Changed.

thomas-zahner commented on lychee-bin/src/formatters/response/mod.rs:
```diff
@@ -25,4 +25,12 @@ pub(crate) const MAX_RESPONSE_OUTPUT_WIDTH: usize = 10;
+    /// Detailed response formatter (defaults to the normal formatter)
+    ///
+    /// This is used for some output modes to provide more detailed information
+    /// and in verbose mode.
```
I find the sentence a bit confusing, especially `and in verbose mode`.

mre replied:
Rephrased. Better now?

thomas-zahner replied:
yes :+1:

thomas-zahner commented on lychee-bin/src/formatters/stats/compact.rs:
```diff
@@ -106,3 +111,76 @@ impl StatsFormatter for Compact {
+
+        assert!(result.contains("🔍 1 Total"));
+        assert!(result.contains("✅ 1 OK"));
+        assert!(result.contains("🚫 0 Errors"));
```
why is it zero errors? :thinking: 
ah `errors` is independent from the error map's size, right? since it's a test it doesn't really matter I guess.

mre replied:
Thanks. Fixed to 2 for now.

mre committed:
fix formatting test

mre committed:
Box Octocrab error as it is too large

thomas-zahner committed:
Add OWASP Threat Dragon to users

mre commented:
@dmathieu, do you have an idea why the test fails..

xlai89 committed:
Don't check prefix attribute (#1536)

thomas-zahner committed:
Add interesting projects who use lychee (#1537)

* Add interesting projects who use lychee

* List Gradle as user in README.md

devin-purple committed:
Fix example usage of headers in configuration file (#1540)

dmathieu commented:
> do you have an idea why the test fails..

It's a bit weird. I'm not getting a failure locally. As if sometimes, a 200 is returned (giving the error we see), and sometimes a 204.

thomas-zahner committed:
Update link check action (#1542)

mre committed:
Add support for relative links (#1489)

This commit introduces several improvements to the file checking process and URI handling:

- Extract file checking logic into separate `Checker` structs (`FileChecker`, `WebsiteChecker`, `MailChecker`)
- Improve handling of relative and absolute file paths
- Enhance URI parsing and creation from file paths
- Refactor `create_request` function for better clarity and error handling

These changes provide better support for resolving relative links, handling different base URLs, and working with file paths.

Fixes https://github.com/lycheeverse/lychee/issues/1296 and https://github.com/lycheeverse/lychee/issues/1480

mre committed:
Fix skipping of email addresses in stylesheets (#1546)

mre committed:
Fix format option in configuration file (#1547)

mre committed:
Refactor cache handling test to make it more robust (#1548)

mre commented:
See #1553 for a cleaned up version of this.
